src/libraries/System.Net.Http/tests/StressTests/HttpStress/ClientOperations.cs

```diff
@@ -318,6 +318,32 @@ public static class ClientOperations
                             }
                         }
 
+                        if (ctx.HttpVersion == HttpVersion.Version30)
+                        {
+                            // HTTP/3 exception nesting:
+                            // HttpRequestException->IOException->HttpRequestException->QuicStreamAbortedException
+                            // HttpRequestException->QuicStreamAbortedException
+
+                            if (e is IOException && e.InnerException is HttpRequestException)
+                            {
+                                e = e.InnerException;
+                            }
+
+                            if (e is HttpRequestException)
+                            {
+                                string? name = e.InnerException?.GetType().Name;
+                                switch (name)
+                                {
+                                    case "QuicStreamAbortedException":
+                                        if (e.InnerException?.Message?.Equals("Stream aborted by peer (258).") ?? false) // 258 = H3_INTERNAL_ERROR (0x102)
```

Comment:
Once we finally clean up the error reporting, we should just check for status within the message as we do with H/2 since this is fragile. But it's good enough for now!

Reply:
I totally agree 😊